### PR TITLE
JSON - Extend Files.GetDirectory method by adding "userrating" to Lis…

### PR DESCRIPTION
**Pull request by scott967**
…t.Fields.Files

## Description

Add the "userrating" property to JSON List.Fields.Files so that it can be retrieved by the  Files.GetDirectory method

## Motivation and context

Currently using Files.GetDirectory in a python addon to pass info to a skin from smart playlists in order to populate a widget panel.  That method does not provide "userrating",  for video, although most everything else is available.

## How has this been tested?

Tested on Windows 10 x64 using Kodi in VS2017 debug.  Modified python addon to request userrating and confirmed that it is available for video in movie smart playlists.

## What is the effect on users?

Used by JSON-RPC consumers.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by ksooo** [comment]
@DaveTBlake any opions on this one?

File: xbmc/interfaces/json-rpc/schema/types.json [hunk 1]
```diff
@@ -1612,7 +1612,7 @@
   "List.Fields.Files": {
     "extends": "Item.Fields.Base",
     "items": { "type": "string",
-      "enum": [ "title", "artist", "albumartist", "genre", "year", "rating",
+      "enum": [ "title", "artist", "albumartist", "genre", "year", "rating", "userrating",
```
Comment (DaveTBlake) [inline xbmc/interfaces/json-rpc/schema/types.json:1615]: Please append "userrating" to the end of the list of enum values after "albumduration" rather than insert.
While I totally get why you would place this next to "rating", but appending new values is the prefered approach.

**Review by DaveTBlake — CHANGES_REQUESTED** [review]
Looks like this was overlooked at some point. Bump of minor version is correct as this is a non-breaking  interface change. Thanks @scott967

File: xbmc/interfaces/json-rpc/schema/types.json (hunk 1 above)
Comment (scott967) [reply xbmc/interfaces/json-rpc/schema/types.json]: Done

**Comment by scott967** [comment]
Failure to build windows seems to be Jenkins didn't compile ffmpeg 4.4?

**Review by DaveTBlake — APPROVED** [review]

**Comment by DaveTBlake** [comment]
jenkins build this please

**Comment by Tolriq** [comment]
Just seeing this, seems it was broken before on the other type but adding "userrating" to the field list is not enough, it should also be present as in the item list as the return value with the proper type.

Else this means the API returns fields that are not defined by the API and so can't know the type for all that generate automatic mapping and everything.

TL;DR:
userrating should be properly defined in "List.Item.All" and "List.Item.File" or all put at Item.xxx.Base

@DaveTBlake

**Comment by Tolriq** [comment]
Test ping @scott967 for previous comment to finish this PR properly :)

Would allow to work from JSON and fix my long lasting todo (`rating = item.rating // TODO: userrating when exposed`)
